src/hypertrace/agent/config/__init__.py

```diff
@@ -128,89 +119,62 @@ def __init__(self):  # pylint: disable=R0912,R0915
-                    if key == 'resource_attributes':
-                        continue
-                    self.validate_config_elements(config_element[key],
-                                                  getattr(agent_config_base, key))  # pylint: disable=W0123
-                    continue
-                except AttributeError as err:
-                    logger.error('Unknown attribute encountered: exception=%s, stacktrace=%s',
-                                 err,
-                                 traceback.format_exc())
-                    continue
-            elif isinstance(config_element[key], (str, bool, int, list)):
-                if key in PYTHON_SPECIFIC_ATTRIBUTES:
-                    logger.debug(
-                        'Found pythonagent-specific attribute, attr=%s', key)
-                    continue
-                try:
-                    if hasattr(agent_config_base, key):
-                        #                      and key != 'propagation_formats':
-                        logger.debug('Is valid: %s', key)
-                    else:
-                        logger.debug('Not valid: %s', key)
-                        raise AttributeError
-                except AttributeError as err:
-                    logger.error('Unknown attribute %s encountered: exception=%s, stacktrace=%s',
-                                 key,
-                                 err,
-                                 traceback.format_exc())
-            else:
-                logger.error('Unknown attribute type encountered: exception=%s, stacktrace=%s',
-                             err,
-                             traceback.format_exc())
+        self.agent_config.enabled = config_dict['enabled']
+
+        self.agent_config.resource_attributes = config_dict['resource_attributes']
 
     def dump_config(self):
         '''Dump configuration information.'''
         logger.debug(self.__dict__)
 
-    def get_config(self) -> config_pb2.AgentConfig:
-        '''Return configuration information.'''
-        return self.agent_config
-
     def use_console_span_exporter(self) -> bool:
         '''Initialize InMemorySpanExporter'''
-        return self.config['_use_console_span_exporter']
+        return self.custom_config.get('_use_console_span_exporter')
+
+
+def _transform_values(config_dict):
+    # This has to be upcase or else enum lookup fails
+    config_dict['reporting']['trace_reporter_type'] = config_dict['reporting']['trace_reporter_type'].upper() # pylint:disable=C0301
+    config_dict['propagation_formats'] = list(map(str.upper, config_dict['propagation_formats']))
+
+def _apply_custom_config_options(current_custom, next_config):
+    for key in PYTHON_SPECIFIC_ATTRIBUTES:
+        if key in next_config:
+            value = next_config.pop(key)
+            current_custom[key] = value
+    return current_custom
+
+def _read_from_file():
+    if 'HT_CONFIG_FILE' not in os.environ:
```

Comment:
Shall this prefix be hardcoded?

Reply:
Merge order makes this seem odd, but the other pr resolved this:
https://github.com/hypertrace/pythonagent/blob/main/src/hypertrace/agent/config/__init__.py#L168